### ISSUE TITLE
Default themes aren't rendering properly for Mapbox

### ISSUE THREAD

**Issue by MendleM** [greensopinion/flutter-vector-map-tiles#125]
Park greenery, dashed paths, and roads don't seem to be styled correctly. Using the streets-v12 Mapbox theme, here's how it looks in their Web view:

![image](https://user-images.githubusercontent.com/20160314/221331772-58532627-d26a-457a-8f6c-23da032a141d.png)

Here's how it looks in the app:

![image](https://user-images.githubusercontent.com/20160314/221331799-f953c26b-09fd-4a47-8a98-916657f0e7f1.png)

Here are all the theme errors I'm seeing in the debug console:

```
I/flutter (16570): WARN: theme layer type circle not implemented
I/flutter (16570): WARN: Unsupported expression syntax: [to-boolean, [coalesce, [image, [concat, [get, shield_beta], -, [to-string, [get, reflen]]]], ]]
I/flutter (16570): WARN: Unsupported expression syntax: [all, [has, shield_text_color_beta], [to-boolean, [coalesce, [image, [concat, [get, shield_beta], -, [to-string, [get, reflen]]]], ]]]
I/flutter (16570): WARN: Unsupported expression syntax: [case, [all, [has, shield_text_color_beta], [to-boolean, [coalesce, [image, [concat, [get, shield_beta], -, [to-string, [get, reflen]]]], ]]], [match, [get, shield_text_color_beta], white, hsl(0, 0%, 100%), yellow, hsl(50, 100%, 70%), orange, hsl(25, 100%, 75%), blue, hsl(230, 57%, 44%), red, hsl(0, 87%, 59%), green, hsl(140, 74%, 37%), hsl(230, 18%, 13%)], [match, [get, shield_text_color], white, hsl(0, 0%, 100%), yellow, hsl(50, 100%, 70%), orange, hsl(25, 100%, 75%), blue, hsl(230, 57%, 44%), red, hsl(0, 87%, 59%), green, hsl(140, 74%, 37%), hsl(230, 18%, 13%)]]
I/flutter (16570): WARN: Unsupported expression syntax: [sqrt, [get, sizerank]]
I/flutter (16570): WARN: Unsupported expression syntax: [-, 16, [sqrt, [get, sizerank]]]
I/flutter (16570): WARN: Unsupported expression syntax: [*, [-, 16, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression: [*, [-, 16, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression syntax: [sqrt, [get, sizerank]]
I/flutter (16570): WARN: Unsupported expression syntax: [-, 22, [sqrt, [get, sizerank]]]
I/flutter (16570): WARN: Unsupported expression syntax: [*, [-, 22, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression: [*, [-, 22, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression syntax: [sqrt, [get, sizerank]]
I/flutter (16570): WARN: Unsupported expression syntax: [-, 16, [sqrt, [get, sizerank]]]
I/flutter (16570): WARN: Unsupported expression syntax: [*, [-, 16, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression: [*, [-, 16, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression syntax: [sqrt, [get, sizerank]]
I/flutter (16570): WARN: Unsupported expression syntax: [-, 22, [sqrt, [get, sizerank]]]
I/flutter (16570): WARN: Unsupported expression syntax: [*, [-, 22, [sqrt, [get, sizerank]]], 1]
I/flutter (16570): WARN: Unsupported expression: [*, [-, 22, [sqrt, [get, sizerank]]], 1]
```

I'm sourcing the theme like this:
```dart 
  void _initMapStyle() async {
    final style = await StyleReader(
      uri: 'mapbox://styles/mapbox/streets-v12/?access_token={key}',
      apiKey: mapBoxApiKey,
      logger: const Logger.console(),
    ).read();

    setState(() => _style = style);
  }
```

And using it like this:
```dart
  Widget _vectorTileLayerBuilder() {
    if (_style == null) {
      return const Center(child: CircularProgressIndicator());
    }

    return VectorTileLayer(
      theme: _style!.theme,
      backgroundTheme: _style!.theme,
      tileOffset: TileOffset.mapbox,
      tileProviders: _style!.providers,
    );
  }

  @override
  Widget build(BuildContext context) {
    return FlutterMap(
      mapController: _controller,
      options: MapOptions(
        center: LatLng(-37.8304837, 144.9712273),
        zoom: 16,
        interactiveFlags: InteractiveFlag.pinchZoom |
            InteractiveFlag.drag |
            InteractiveFlag.doubleTapZoom |
            InteractiveFlag.flingAnimation,
      ),
      nonRotatedChildren: [_attributions()],
      children: [
        _vectorTileLayerBuilder(),
      ],
    );
  }
```

**Comment by greensopinion** [comment]
The log indicates that some style expressions aren't yet supported by this library.
Here's an example of adding support for an expression: [dart-vector-tile-renderer/pull/69](https://github.com/greensopinion/dart-vector-tile-renderer/pull/69)
Feel free to make a contribution!

**Comment by greensopinion** [comment]
This is fixed in version 3.2.0 of [vector_tile_renderer](https://pub.dev/packages/vector_tile_renderer)
